### ISSUE TITLE
Magento\Email\Model\Template\Filter can't be intercepted

### ISSUE THREAD

**Issue by kassner** [magento/magento2#4648]
Hi,

Magento version: 2.0.6

I created a simple plugin: (class exists, but its contents does not interfere here).

``` xml
<type name="\Magento\Email\Model\Template\Filter">
    <plugin name="escapenl2br" type="Vendor\Module\Plugin\EscapeNl2br" sortOrder="1"/>
</type>
```

But this results in a weird error message:

```
Warning: Declaration of Magento\Email\Model\Template\Filter\Interceptor::setTemplateProcessor($callback) should be compatible with Magento\Framework\Filter\Template::setTemplateProcessor(callable $callback) in /private/var/www/magento/var/generation/Magento/Email/Model/Template/Filter/Interceptor.php on line 7
```

Obviously, the code generated is wrong:

``` php
    public function setTemplateProcessor($callback)
    {
        $pluginInfo = $this->pluginList->getNext($this->subjectType, 'setTemplateProcessor');
        if (!$pluginInfo) {
            return parent::setTemplateProcessor($callback);
        } else {
            return $this->___callPlugins('setTemplateProcessor', func_get_args(), $pluginInfo);
        }
    }
```

Where I can dig into to fix this?

Thanks!

**Comment by basselalaraaj** [comment]
This seems already fixed in:
https://github.com/magento/magento2/blob/develop/lib/internal/Magento/Framework/Code/Generator/EntityAbstract.php::328 / 330

**Comment by p-bystritsky** [comment]
@kassner, thank you for your report.
The issue is already fixed in 2.1.10, 2.2.1 and 2.3.0.

**Comment by magento-engcom-team** [comment]
@kassner, thank you for your report.
The issue is already fixed in 2.1.10, 2.2.1 and 2.3.0.

**Comment by magento-engcom-team** [comment]
@kassner, thank you for your report.
The issue is already fixed in develop branch, 2.1.10, 2.2.1
